pkg/flow/internal/controller/component_node.go

```diff
@@ -0,0 +1,46 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/grafana/agent/component"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ComponentInfo is an interface that encapsulates methods for managing and retrieving detailed information about a component within a BlockNode.
+type ComponentInfo interface {
```

Comment:
Unfortunately I also have some issues with this interface that I think should be resolved:

* I think the name and description doesn't really represent my mental modal of how nodes in the DAG work. Rather than a component being "within" a BlockNode, I see a node presenting itself _as_ a component. 

* I think this interface is very broad, and probably inclusive of things that don't make sense for non-builtin components: at first glance `DebugInfo()`, `Component()`, `ModuleIDs()`, and `Registry()` seem to be specific to builtin components. 

* `BlockName()` throws me off a little. I know it's literally what is being returned, but it can be seen as a leaky abstraction for the configuration language and where the component name may come from. I would rather this be called `ComponentName()`, which more representative of the data being returned and is named agnostically to how that data gets populated in the node. 

I'd like to see these changes:

1. Rename the `ComponentNode` struct to `BuiltinComponentNode`, and allow this interface to be `ComponentNode`. 

2. Minimize the interface as much as possible. Only methods which make sense for _all_ possible implementations of a component should remain. 

3. Rename `BlockName()` to `ComponentName()`. 

WDYT?

Reply:
In the current implementation I already have an interface for ComponentNode. I wanted to bring it with the rename of the ComponentNode to BuiltinComponentNode via a different PR.
- BuiltinComponentNode / CustomComponentNode implement Component Node (used by the Loader to handle both as ComponentNode)
- BuiltinComponentNode / CustomComponentNode / ImportConfigNode implement Component Info (used by the component provider to retrieve information about running components)

How should we handle the naming?

About removing the functions I think that makes sense, I can use type switch in the getComponentDetail to pass the builtin component info to build the component.Info

Reply:
But do we want the Import node to appear in the UI? It was a bit of a stretch to pass it as Component because it contains a component but at least it would reuse the current logic to pass the data through the API

Reply:
> But do we want the Import node to appear in the UI? It was a bit of a stretch to pass it as Component because it contains a component but at least it would reuse the current logic to pass the data through the API


For now I'm OK with the import node _not_ appearing in the UI. The UI needs improvements in general (such as showing configuration blocks), so we'll find the best way to expose other non-component nodes when it's time to expand the UI to cover more than components.

Reply:
I made the changes and I also put the block node interface inside of the runnable interface to follow the schema that you put above. I think that this hierarchy makes sense and makes it easier to follow